exporter/clickhouseexporter/internal/sum_metrics.go

```diff
@@ -0,0 +1,171 @@
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+	ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+    TimeUnix,
+    ValueAsDouble,
+    ValueAsInt,
+    Flags,
+    Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId,
+	AggTemp,
+	IsMonotonic) VALUES `
+	sumPlaceholders = "(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)"
+)
+
+type sumModel struct {
+	metricName        string
+	metricDescription string
+	metricUnit        string
+	sum               pmetric.Sum
+}
+
+type SumMetrics struct {
+	sumModel  []*sumModel
+	metadata  *MetricsMetaData
+	insertSQL string
+}
+
+func (s *SumMetrics) insert(ctx context.Context, tx *sql.Tx, logger *zap.Logger) error {
+	var valuePlaceholders []string
+	var valueArgs []interface{}
+
+	for _, model := range s.sumModel {
+		for i := 0; i < model.sum.DataPoints().Len(); i++ {
+			dp := model.sum.DataPoints().At(i)
+			valuePlaceholders = append(valuePlaceholders, sumPlaceholders)
+
+			valueArgs = append(valueArgs, s.metadata.ResAttr)
+			valueArgs = append(valueArgs, s.metadata.ResURL)
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.Name())
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.Version())
+			valueArgs = append(valueArgs, attributesToMap(s.metadata.ScopeInstr.Attributes()))
+			valueArgs = append(valueArgs, s.metadata.ScopeInstr.DroppedAttributesCount())
+			valueArgs = append(valueArgs, s.metadata.ScopeURL)
+			valueArgs = append(valueArgs, model.metricName)
+			valueArgs = append(valueArgs, model.metricDescription)
+			valueArgs = append(valueArgs, model.metricUnit)
+			valueArgs = append(valueArgs, attributesToMap(dp.Attributes()))
+			valueArgs = append(valueArgs, dp.Timestamp().AsTime().UnixNano())
+			valueArgs = append(valueArgs, dp.DoubleValue())
+			valueArgs = append(valueArgs, dp.IntValue())
+			valueArgs = append(valueArgs, uint32(dp.Flags()))
+
+			attrs, times, floatValues, intValues, traceIDs, spanIDs := convertExemplars(dp.Exemplars())
+			valueArgs = append(valueArgs, attrs)
+			valueArgs = append(valueArgs, times)
+			valueArgs = append(valueArgs, floatValues)
+			valueArgs = append(valueArgs, intValues)
+			valueArgs = append(valueArgs, traceIDs)
+			valueArgs = append(valueArgs, spanIDs)
+			valueArgs = append(valueArgs, int32(model.sum.AggregationTemporality()))
+			valueArgs = append(valueArgs, model.sum.IsMonotonic())
+		}
+	}
+
+	if len(valuePlaceholders) == 0 {
+		return nil
+	}
+
+	start := time.Now()
+	_, err := tx.ExecContext(ctx, fmt.Sprintf("%s %s", s.insertSQL, strings.Join(valuePlaceholders, ",")), valueArgs...)
+	if err != nil {
+		return fmt.Errorf("insert sum metrics fail:%w", err)
+	}
+	duration := time.Since(start)
+
+	// TODO latency metrics
+	logger.Debug("insert sum metrics", zap.Int("records", len(valuePlaceholders)),
+		zap.String("cost", duration.String()))
```

Comment:
I think zap can report a time, no need to convert to string

Reply:
done